### PR TITLE
rds: do not update multi-az unless specified - fixes #23385

### LINKED ISSUE #23385: rds.py reverts multi-AZ RDS back to single-AZ on any 'modify' action where 'multi_zone' is not specified
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
rds.py

##### ANSIBLE VERSION
```
ansible 2.3.0.0 (stable-2.3 0fe53eda86) last updated 2017/04/06 22:20:16 (GMT -500)
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Dec 18 2016, 07:03:39) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]

```

##### CONFIGURATION
n/a

##### OS / ENVIRONMENT
n/a

##### SUMMARY
Any 'modify' action using the rds.py module taken against a multi-AZ RDS, where the optional `multi_zone` parameter is not explicitly included in the task, results in the RDS being modified to a single-AZ RDS.

##### STEPS TO REPRODUCE
Example playbook:

```
---

- name: Update password on RDS
  hosts: localhost
  tasks:
    - name: Apply new password to RDS
      rds:
        region: us-east-1
        command: modify
        instance_name: some-rds-12345
        password: supersekrit123

```

##### EXPECTED RESULTS
The `password` and/or other specified parameters are updated, while the `multi_zone` (multi-AZ) status of the RDS is left alone unless specified.

##### ACTUAL RESULTS
The requested parameters are updated but in addition, the RDS is changed as if we had specified `multi_zone` to false (ie. multi-AZ is disabled).

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
multi_zone [defaults to False](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/rds.py#L1084) rather than [None as the documentation states](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/rds.py#L108) so [this line](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/rds.py#L1040) does not work as expected when multi-az isn't specified. Other parameters are not updated unless they are specified explicitly. Fixes #23385

Also fixed rds to be pep8 and removed from legacy files.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon/rds.py

##### ANSIBLE VERSION
```
2.4.0
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/rds.py [hunk 1]
```diff
@@ -1124,6 +1124,9 @@ def main():
         'restore': restore_db_instance,
     }
 
+    if module.params.['command'] != 'modify' and module.params.['multi_zone'] is None:
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/rds.py]: shouldn't this just be `module.params['command']`? (without the second `.`)

File: lib/ansible/modules/cloud/amazon/rds.py [hunk 2]
```diff
@@ -1124,6 +1124,9 @@ def main():
...
+        module.params.['multi_zone'] = False
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/rds.py]: Having said that, I don't believe these two lines are required at all - we should only use multi_zone if it's explicitly set, and let the boto rds2 API handle the default (which it does - if you pass it multi_az=None (it's default), you'll get a non multi AZ RDS instance)

**Comment by s-hertel** [comment]
@willthames I think you must have submitted your review after I corrected myself. How does this look to you?

File: lib/ansible/modules/cloud/amazon/rds.py (hunk 2 above)
Comment (s-hertel) [reply lib/ansible/modules/cloud/amazon/rds.py]: @willthames Oh, good point. Thanks! Fixed it.

**Pull request merged**